File: 27-RemoveElement/27-RemoveElement.py
# Last updated: 8/5/2025, 12:36:48 PM
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n=len(nums)
        i=0
        while i<n:
            if nums[i] == val: 
                nums.pop(i)
                n-=1
            else:
                i+=1
        return n
        

File: 27-RemoveElement/27-RemoveElement.java
// Last updated: 8/5/2025, 12:36:44 PM
// Using fast-slow pointer approach
class Solution {
    public int removeElement(int[] nums, int val) {
        int fast = 0; //pointer to iterate through every element
        int slow = 0; // track the position where the next valid element should go
        for (fast = 0; fast < nums.length; fast++) {
            if (nums[fast] != val) {
                nums[slow] = nums[fast];
                slow++;
            }
        }
        return slow;
    }
}
